### ISSUE TITLE
How to deal with converting a large onnx model to TensorRT engine?

### ISSUE THREAD

**Issue by chiehpower** [NVIDIA/TensorRT#1348]
## Description

I have a very large onnx model (almost over 10GB) that I wanna convert it to TensorRT engine.
However, when I converted the onnx model with external data to TensorRT engine, it encountered the error below:

```
&&&& RUNNING TensorRT.sample_onnx # ./sample_onnx
DataDirs Area: add the data directory.
[07/06/2021-15:44:36] [I] Building and running a GPU inference engine
Model path: /workspace/model_external_10GB.trt
[07/06/2021-15:44:36] [I] 
[07/06/2021-15:44:36] [I] >> Start to generate the model.
----------------------------------------------------------------
Input filename:   /workspace/model/model.onnx
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    pytorch
Producer version: 1.7
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
[07/06/2021-15:45:09] [W] [TRT] [TRT]/home/jenkins/workspace/OSS/L0_MergeRequest/oss/parsers/onnx/onnx2trt_utils.cpp:220: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[07/06/2021-15:45:10] [E] [TRT] Tensor: (Unnamed Layer* 125) [Constant]_output at max batch size of 1 exceeds the maximum element count of 2147483647
[07/06/2021-15:45:10] [E] [TRT] Network validation failed.
&&&& FAILED TensorRT.sample_onnx # ./sample_onnx
```
This one was written by C++ TensorRT API to generate the TensorRT model from Onnx.
In addition, I can use this code to generate the TensorRT engine with a smaller onnx model with external data (1.1GB), so I think that the code was working well.
 
## Environment

**TensorRT Version**: 7.1
**NVIDIA GPU**: Titan V
**NVIDIA Driver Version**: 450 
**CUDA Version**: 10.2
**Operating System**: ubuntu 1804
**Python Version (if applicable)**:  3.6
**PyTorch Version (if applicable)**: 1.7

## Relevant Files

Sorry, the onnx model is over 10GB that it is hard to share.

I wonder is there anyone who successfully converted a very large model to TensorRT engine before?
Is there any relevant document?

Thank you so much!!!

Best regards,
Chieh

**Comment by ttyio** [comment]
Sorry @chiehpower today TRT support tensor at most 2G elements. What's in this constant layer, is this an embedding table? Thanks

**Comment by chiehpower** [comment]
Hi @ttyio,

Thanks for your reply! So sad to get this information!

Indeed, I just resolved the issue to implement the onnx model which overtakes 2GB to do inference properly.

The onnx model is saved without external data. Hence, it can highly reduce the onnx model, and in the same folder, there are some relevant data. 

The model is only few conv layers and the others are data (or knowledge) that its size depends on the parameter setting.
I have done some experiments that I used a smaller model (same structure) to compare the inference speed between onnxruntime and TensorRT, and TensorRT did not get a significant improvement. (only faster than using onnxruntime around a few ms) So I think that probably my case is not really suitable for using TensorRT.

Anyway, thanks again!

BR,
Chieh
